packages/ses/src/tame-global-error-object.js

```diff
@@ -97,6 +146,39 @@ export default function tameGlobalErrorObject(errorTaming = 'safe') {
       enumerable: false,
       configurable: true,
     },
+    prepareStackTrace: {
+      get() {
+        if (
+          errorTaming === 'unsafe' &&
+          typeof userPrepareStackTrace === 'function'
+        ) {
+          return tamedMethods.prepareStackTrace;
```

Comment:
On non-v8 in "unsafe" mode, `Error.prepareStackTrace` would be just another mutable ignored property, undefined by default. Code that does `Error.prepareStackTrace = 4; Error.prepareStackTrace + 1` should get 5. Should we try to emulate that? Or is it too hard/dangerous to distinguish between a user reading this property and the VM itself? I don't even know how we'd sense that we're on an engine that treats `prepareStackTrace` as special.. maybe try it at startup and see if it gets called? Ugh.

Reply:
I think this will also not emulate v8's behavior if you set `Error.prepareStackTrace = 4` (and then read it or set it back again carefully before creating an Error or reading it's `.stack`).

Also, does `delete Error.prepareStackTrace` behave like `Error.prepareStackTrace = undefined`?

Reply:
Also, as we discussed, I don't think this will nest correctly (and I don't see a test of that yet).

```js
let stackA = Error.captureStackTrace();
let old1 = Error.prepareStackTrace;
Error.prepareStackTrace = new1;
let stackB = Error.captureStackTrace();
let old2 = Error.prepareStackTrace; // here
Error.prepareStackTrace = new2;
let stackC = Error.captureStackTrace();
Error.prepareStackTrace = old2; // there
let stackD = Error.captureStackTrace();
Error.prepareStackTrace = old1;
let stackE = Error.captureStackTrace();
// A and E should get the built-in default behavior
// B and D should get 'new1' behavior
// C should get 'new2' behavior
```

The line marked `here` will set `old2` to the wrapper, so the line marked `there` will set `userPrepareStackTrace` to the wrapper too, which will cause an infinite loop when finally executed.

I'm not sure how to fix this. But I'd start with a (disabled) test demonstrating the problem.

Reply:
> Should we try to emulate that?

IMO no. The value read will differ from the value written when they are functions. We accept the assignment only because of the "compatibility" advice paragraph at https://v8.dev/docs/stack-trace-api#compatibility . These properties remain magical enough that I am only concerned with accommodating cases we actually encounter, since we cannot emulate non-ses behavior at high fidelity anyway.

Reply:
> I think this will also not emulate v8's behavior if you set Error.prepareStackTrace = 4 (and then read it or set it back again.

Correct. For functions, reading gives back a wrapping function. For non-functions, reading gives `undefined`.

Reply:
> Also, as we discussed, I don't think this will nest correctly (and I don't see a test of that yet).

Test added at the end of error-manipulation.test.js . I added the test first and saw if fail as expected. I added the test before seeing your example code above. Your code is technically incorrect in that `captureStackTrace` only mutates its arguments. It returns `undefined`. That aside, I think my test captures everything in yours. PTAL.

Reply:
The infinite recursion was indeed there and is now fixed.